addons/sale_purchase_stock/tests/test_unwanted_replenish_flow.py

```diff
@@ -0,0 +1,279 @@
+from datetime import datetime, timedelta
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from odoo.tests import common, Form, tagged
+
+@tagged('post_install', '-at_install')
+class TestWarnUnwantedReplenish(common.TransactionCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.buy_route = cls.env.ref('purchase_stock.route_warehouse0_buy')
+
+         # Create a vendor (& suppliers) and a customer
+        cls.vendor = cls.env['res.partner'].create(dict(name='Vendor'))
+        cls.customer = cls.env['res.partner'].create(dict(name='Customer'))
+
+        cls.supplier_A = cls.env['product.supplierinfo'].create({
+            'partner_id' : cls.vendor.id,
+            'min_qty' : 0.0,
+            'price' : 10.0,
+            'delay' : 0
+        })
+
+        cls.supplier_B = cls.env['product.supplierinfo'].create({
+            'partner_id' : cls.vendor.id,
+            'min_qty' : 0.0,
+            'price' : 12.0,
+            'delay' : 0
+        })
+
+        # Create a "A" and a "B" Product :
+        # No Stock
+        # Partner/Customer Lead Time = 0
+        # Manual reordering 0 0
+
```

Comment:
Instead of putting intermediate comment it's better to add a docstring at the beginning of the test